fdbrpc/FlowTests.actor.cpp

```diff
@@ -1271,3 +1272,115 @@ TEST_CASE("/flow/DeterministicRandom/SignedOverflow") {
 	       std::numeric_limits<int64_t>::max() - 1);
 	return Void();
 }
+
+class PrivateKeyRangeTestImpl : public PrivateKeyRangeBaseImpl {
+public:
+	explicit PrivateKeyRangeTestImpl(KeyRef start, KeyRef end, std::string prefix, int size) : PrivateKeyRangeBaseImpl(start, end) {
+		this->prefix = prefix;
+		this->size = size;
+		ASSERT(size > 0);
+		for (int i = 0; i < size; ++i) {
+			kvs.push_back_deep(kvs.arena(), KeyValueRef(getKeyForIndex(i), deterministicRandom()->randomAlphaNumeric(16)));
+		}
+	}
+
+	KeyValueRef getKeyValueForIndex(int idx) {return kvs[idx];}
+
+	Key getKeyForIndex( int idx ) {
+		return Key( prefix + format("%010d", idx)).withPrefix(range.begin);
+	}
+
+	virtual Future<Standalone<RangeResultRef>> getRange(ReadYourWritesTransaction* ryw, KeyRangeRef kr) const override {
+		int startIndex=0, endIndex= size;
+		while (startIndex < size && kvs[startIndex].key < kr.begin)
+			++startIndex;
+		while (endIndex > startIndex && kvs[endIndex-1].key >= kr.end)
+			--endIndex;
+		if (startIndex == endIndex)
+			return Standalone<RangeResultRef>();
+		else
+			return Standalone<RangeResultRef>(RangeResultRef(kvs.slice(startIndex, endIndex), false));
+	}
+private:
+	Standalone<VectorRef<KeyValueRef>> kvs;
+	std::string prefix;
+	int size;
+};
+
+TEST_CASE("/fdbclient/PrivateKeySpace/Aggregation") {
+	PrivateKeySpace pks(LiteralStringRef("\xff\xff"), LiteralStringRef("\xff\xff\xff"));
+	PrivateKeyRangeTestImpl pkr1(LiteralStringRef("\xff\xff/cat/"), LiteralStringRef("\xff\xff/cat/\xff"), "small", 10);
+	PrivateKeyRangeTestImpl pkr2(LiteralStringRef("\xff\xff/dog/"), LiteralStringRef("\xff\xff/dog/\xff"), "medium", 100);
+	PrivateKeyRangeTestImpl pkr3(LiteralStringRef("\xff\xff/pig/"), LiteralStringRef("\xff\xff/pig/\xff"), "large", 1000);
+	pks.registerKeyRange(pkr1.getKeyRange(), &pkr1);
+	pks.registerKeyRange(pkr2.getKeyRange(), &pkr2);
+	pks.registerKeyRange(pkr3.getKeyRange(), &pkr3);
+	// get
+	{
+		auto resultFuture = pks.get(NULL, LiteralStringRef("\xff\xff/cat/small0000000009"));
```

Comment:
```suggestion
		auto resultFuture = pks.get(nullptr, LiteralStringRef("\xff\xff/cat/small0000000009"));
```

Reply:
Done. Changed all `NULL` to `nullptr`. Thanks.